GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -225,15 +225,29 @@ protected override void Dispose(bool disposing)
         {
             if (disposing)
             {
+                _superprojectCurrentCheckout = null;
+                _selectedObjectIds = null;
+                _baseCommitToCompare = null;
                 _revisionSubscription?.Dispose();
+                _revisionSubscription = null;
                 _revisionReader?.Dispose();
+                _revisionReader = null;
+                _artificialStatus = null;
+                _lastVisibleResizableColumn?.Dispose();
+                _lastVisibleResizableColumn = null;
+                _maximizedColumn?.Dispose();
+                _selectionTimer?.Dispose();
                 _buildServerWatcher?.Dispose();
 
                 if (_indexWatcher.IsValueCreated)
                 {
                     _indexWatcher.Value.Dispose();
                 }
 
+                _loadingControlSync.Dispose();
+                _loadingControlAsync.Dispose();
+                _revisionFilter?.Dispose();
```

Comment:
❔  Are these not part of `components` where they will get automatically disposed?

Reply:
Yes for `_loadingControl...` but not `_revisionFilter`:
`private readonly FormRevisionFilter _revisionFilter = new FormRevisionFilter();`
I was not disposed, at least not before `AfterTest`.

I wanted to be on the safe side. Setting to `null` and calling `Dispose` twice may be pointless but it shouldn't hurt.
Finally, I found an exception thrown in `Dispose`. This exception was not captured by NUnit.

Reply:
I am going to replace `_field = null;` with `//// _field is not disposable` in order to show that each field has been considered in `Dispose` and maybe also in `TearDown`.

Reply:
Is Re# able to assist with `Dispose`?